src/controllers/user.controllers.ts

```diff
@@ -1,47 +1,73 @@
 import { ValidatedRequest } from 'express-joi-validation';
 import { v4 as uuidv4 } from 'uuid';
-import { getAutoSuggestUsers } from '../helpers';
-import { IUser, QueryParams, RequestSchema } from '../types';
+import { QueryParams, RequestSchema } from '../types';
 import { Request, Response } from 'express';
+import userService from '../services/user.service';
 
-const users: IUser[] = [];
+const genericErrorMessage = 'Error occurred while processing request';
 
-export const createUser = (req: ValidatedRequest<RequestSchema>, res: Response) => {
+export const createUser = async (req: ValidatedRequest<RequestSchema>, res: Response) => {
     const { login, password, age } = req.body;
-    const user = { id: uuidv4(), login, password, age, isDeleted: false };
-    users.push(user);
-    res.json(`User with id ${user.id} has been created`);
+    const user = { id: uuidv4(), login, password, age };
+
+    try {
+        const userRecord = await userService.createUser(user);
```

Comment:
Usually we use 2 space indentation instead of 4 space. Please check the autoformatter.